### ISSUE TITLE
EmptyLineSeparator check does not validate newlines before comments

### ISSUE THREAD

**Issue by vovkab** [checkstyle/checkstyle#2974]
```
$ javac Test.java
$ cat Test.java
public class Test {

    public static final void method1() { }



    // Method 2 must fail
    public static final void method2() { }



    /**
     * Method 3 must fail
     */
    public static final void method3() { }
}


$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="EmptyLineSeparator">
            <property name="allowNoEmptyLineBetweenFields" value="true"/>
            <property name="allowMultipleEmptyLines" value="false"/>
        </module>
    </module>
</module>


$ java -jar checkstyle-6.15-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

**Expected:**
`method2` and `method3` must fail with `... has more than 1 empty lines before. [EmptyLineSeparator]`

Related tickets: #2067 

http://checkstyle.sourceforge.net/config_whitespace.html#EmptyLineSeparator

**Comment by romani** [comment]
please give us more details.
See http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

**Comment by vovkab** [comment]
@romani done

**Comment by romani** [comment]
Please prove what is expected

**Comment by vovkab** [comment]
@romani not sure what do you mean, everything that should fail is in a comments.

**Comment by romani** [comment]
Please put all to this issue descriprion

**Comment by vovkab** [comment]
Not sure why we need to repeat it in multiple places, but anyway, added expected bahavior

**Comment by romani** [comment]
thanks a lot. It helps to make issue compact and independent to any other issues and their context.

**Comment by liscju** [comment]
This issue is more tricky to implement than it seems ,because it contains many rough edges in checking comments text like:

```java
public static final void method1() { } /**

/** Method 2 should not fail
*/
public static final void method2() { }
```
or 
```java
public static final void method1() { } 

// There is empty line before
/** Method 2 should not fail
*/
// Additional comment
public static final void method2() { }
```
or 
```java
public static final void method1() { }  /** Method 2 should fail
*/
// Additional comment
public static final void method2() { }
```

I started implementing it in https://github.com/liscju/checkstyle/commit/d91467b132adda9c9fa216fb3b8c47074e4754e9 , but fixing this in general is implementing text search with many rough edges , i dont know if i will continue to do this - so for now im leaving my thoughts about it :P

**Comment by romani** [comment]
pre-assigned https://groups.google.com/forum/#!topic/checkstyle-devel/ee95NUc09p0

**Comment by shivanshsoni** [comment]
I am on it

**Comment by shivanshsoni** [comment]
check [commit](https://github.com/shivanshsoni/checkstyle/pull/1/commits/5062c726b1f9d34f5bbca6aca951b9afce914200l)

**Comment by shivanshsoni** [comment]
@romani I think it is not that easy to implement this. But I will try for it...
I already solved an issue #3976  and if it is required to solve more than one for GSOC please assign me another issue which can be solved in less time and if not required then I will try this.

**Comment by romani** [comment]
keep focusing on this issue, I do not see it as complicated.
If you fix one complicated issue it is even better than few easy issues.

**Comment by shivanshsoni** [comment]
@romani Ok Sir, I really want to be in GSOC this year that's why I asked about that. I am working day and night to get selected in  GSOC
I am working on above issue and you will see some positive results very soon. :-)

**Comment by jjlharrison** [comment]
I've attempted to fix this issue and I've used some of the test cases in @liscju's [commit](https://github.com/liscju/checkstyle/commit/d91467b132adda9c9fa216fb3b8c47074e4754e9).

However, I've been unable to cover 2 of the conditional branches which is causing the coverage check to fail. I'd appreciate it if someone could look at my pull-request (#5486) and point me in the right direction.

**Comment by romani** [comment]
closed in favor of #5981 , no changes in description just to loose all connections to previous fixes
